File: src/Zorvian.Core/Entities/ServiceProvider.cs
namespace Zorvian.Core.Entities;

public sealed class ServiceProvider : BaseEntity
{
    public Guid CollaboratorId { get; set; }
    public Collaborator Collaborator { get; set; } = null!;

    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; } = null!;

    public string BusinessName { get; set; } = string.Empty;
    public string? FiscalAddress { get; set; }
    public string? TaxRegime { get; set; }
    public string? ProfessionalLicense { get; set; }
    public string? Specialization { get; set; }
    public string ServiceCategory { get; set; } = string.Empty;
    public string? InsurancePolicy { get; set; }
    public DateOnly? InsuranceExpiration { get; set; }
    public string Status { get; set; } = "active";

    public ICollection<ServiceContract> Contracts { get; set; } = [];
}

File: frontend/lib/core/entities/service_provider.dart
class ServiceProvider {
  final String id;
  final String employeeId;
  final String businessName;
  final String? fiscalAddress;
  final String? taxRegime;
  final String? professionalLicense;
  final String? specialization;
  final String serviceCategory;
  final String? insurancePolicy;
  final DateTime? insuranceExpiration;
  final String status;

  ServiceProvider({
    required this.id,
    required this.employeeId,
    required this.businessName,
    this.fiscalAddress,
    this.taxRegime,
    this.professionalLicense,
    this.specialization,
    required this.serviceCategory,
    this.insurancePolicy,
    this.insuranceExpiration,
    this.status = 'active',
  });

  factory ServiceProvider.fromJson(Map<String, dynamic> json) {
    return ServiceProvider(
      id: json['id'],
      employeeId: json['employeeId'],
      businessName: json['businessName'],
      fiscalAddress: json['fiscalAddress'],
      taxRegime: json['taxRegime'],
      professionalLicense: json['professionalLicense'],
      specialization: json['specialization'],
      serviceCategory: json['serviceCategory'],
      insurancePolicy: json['insurancePolicy'],
      insuranceExpiration: json['insuranceExpiration'] != null 
          ? DateTime.parse(json['insuranceExpiration']) 
          : null,
      status: json['status'] ?? 'active',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'employeeId': employeeId,
    'businessName': businessName,
    'fiscalAddress': fiscalAddress,
    'taxRegime': taxRegime,
    'professionalLicense': professionalLicense,
    'specialization': specialization,
    'serviceCategory': serviceCategory,
    'insurancePolicy': insurancePolicy,
    'insuranceExpiration': insuranceExpiration?.toIso8601String().split('T')[0],
    'status': status,
  };
}
